packages/authentication/src/web.ts

```diff
@@ -121,8 +122,23 @@ export class FirebaseAuthenticationWeb
     auth.languageCode = options.languageCode;
   }
 
-  public async signInWithApple(): Promise<SignInResult> {
-    const provider = new OAuthProvider('apple.com');
+  // Attach scopes to provider if they're there.
+  // Takes in either a string describing an OAuthProvider, or specific other providers.
+  private createProviderWithScopes(
+    provider: string | GoogleAuthProvider | FacebookAuthProvider,
+    options?: SignInOptions,
+  ) {
+    if (typeof provider === 'string') provider = new OAuthProvider(provider);
+    if (options?.scopes) {
+      for (const scope of options.scopes) {
+        provider.addScope(scope);
+      }
+    }
+    return provider;
+  }
```

Comment:
I suggest that we proceed here in the same way as above: Revert the changes in this file and create a new method `applySignInConfig(provider: BaseOAuthProvider, options?: SignInOptions)` which is called after you create the provider in each method.